src/main/java/xyz/nkomarn/harbor/util/PlayerManager.java

```diff
@@ -69,13 +79,8 @@ public boolean isAfk(@NotNull Player player) {
             return false;
         }
 
-        Optional<Essentials> essentials = harbor.getEssentials();
-        if (essentials.isPresent()) {
-            User user = essentials.get().getUser(player);
-
-            if (user != null) {
-                return user.isAfk();
-            }
+        if (afkProviders.stream().anyMatch(provider -> provider.isAFK(player))) {
+            return true;
```

Comment:
would be good to define if it should be interpreted as an AND or and OR.
Sometimes you may want a player to be only be flagged as afk when all providers confirm that the player is afk.
Sometimes you may want a player to be flagged as afk when any provider confirms that the player is afk.

Just my two cents

Reply:
That can be a config flag, I'll make an addAsAnd and addAsOr to both providers

Reply:
Your AFK code has been merged to mine, and I put in the ability to select AND or OR.